src/hotspot/share/opto/callnode.hpp

```diff
@@ -913,6 +913,22 @@ class CallLeafNode : public CallRuntimeNode {
 #endif
 };
 
+class CallLeafPureNode : public CallLeafNode {
```

Comment:
You need a short description about what this node is for. What are the assumptions about it?